modules/testkit/shared/src/main/scala/gsp/math/arb/ArbPlace.scala

```diff
@@ -14,24 +20,29 @@ trait ArbPlace {
   import ArbDeclination._
   import ArbTime._
 
-  private val MinAltitude: Double = 0.0
-  private val MaxAltitude: Double = 8000.0
+  private val MinAltitude: Int = 0
+  private val MaxAltitude: Int = 8000
 
-  val genEarthAlt: Gen[Double] =
-    Gen.chooseNum(MinAltitude, MaxAltitude)
+  val genEarthAlt: Gen[NonNegInt] =
+    Gen.chooseNum(MinAltitude, MaxAltitude).map { a =>
+      refineV[NonNegative](a)
+        .getOrElse(sys.error("Min altitude is non negative"))
+    }
```

Comment:
Maybe we can use `Gen(..).suchThat` here.

Reply:
I didn't know that construct seems good though it is probably not applicable in this case.
`refineV` has to be called on the value and it returns an `Either`

Reply:
Ah, does refineV transform the value or just checks it's non negative? If it's the latter we can use .suchThat{ a => refineV[NonNegative](a).isRight} ?

Reply:
That was my first instinct but (if I understand `suchThat` it will produce an `A` rather than a `A Refined NonNegative`